src/modules/dashboard/components/SmiteDialog/SmiteDialogForm.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useMemo, useCallback, ReactNode } from 'react';
+import { FormikProps } from 'formik';
+import { defineMessages } from 'react-intl';
+import sortBy from 'lodash/sortBy';
+
+import Button from '~core/Button';
+import { ItemDataType } from '~core/OmniPicker';
+import { ActionDialogProps } from '~core/Dialog';
+import DialogSection from '~core/Dialog/DialogSection';
+import { Select, Input, Annotations, SelectOption } from '~core/Fields';
+import Heading from '~core/Heading';
+import SingleUserPicker, { filterUserSelection } from '~core/SingleUserPicker';
+
+import { Address } from '~types/index';
+import HookedUserAvatar from '~users/HookedUserAvatar';
+import { AnyUser, OneDomain } from '~data/index';
+
+import { FormValues } from './SmiteDialog';
+import TeamDropdownItem from './TeamDropdownItem';
+
+import styles from './SmiteDialogForm.css';
+
+const MSG = defineMessages({
+  title: {
+    id: 'dashboard.SmiteDialog.SmiteDialogForm.title',
+    defaultMessage: 'Smite',
+  },
+  team: {
+    id: 'dashboard.SmiteDialog.SmiteDialogForm.team',
+    defaultMessage: 'Select Team in which punishment should happen',
```

Comment:
It doesn't sound very correct either. My suggestion would be:
`Select a Team where the punishment should be applied`
Again, I was running it by my native speaker friend.